fedimint-api-client/src/api/mod.rs

```diff
@@ -940,7 +941,12 @@ where
 {
     #[instrument(level = "trace", fields(peer = %self.peer_id, %method), skip_all)]
     pub async fn request(&self, method: &str, params: &[Value]) -> JsonRpcResult<Value> {
-        for attempts in 0.. {
+        for (attempts, retry_delay) in backon::FibonacciBuilder::default()
+            .with_min_delay(Duration::from_millis(200))
+            .with_max_delay(Duration::from_secs(5))
+            .build()
+            .enumerate()
+        {
             debug_assert!(attempts <= 1);
```

Comment:
no, this loop doesn't really loop

there should be some backoff in reconnect.

Reply:
![image](https://github.com/fedimint/fedimint/assets/9209/51996eb7-dbaf-46f6-8029-ecf321b1cc79)

Reply:
> at most it will do one more round to re-try after reconnecting once

In optimized builds, the `debug_assert` is removed. Just want to be sure that isn't what's causing this issue. @maan2003 the hammering issue occurs on a dev build, right?

Apologies if this is off-track, I'm still trying to understand what this code is doing lol

Reply:
@maan2003 @dpc I remember us also stumbling over this loop, we should think about a more understandable version of it

Reply:
I actually want this loop to actually loop one day. BTW. IIRC we did have strategies that relied on it returning after first disconnect, but it seemed to me somewhat misguided and artificial. 

This loop looping or not has nothing to do with backoffs anyway. The backoff itself is faciliated by the actual underlying client being wrapped in `Jit` that has sleep in it, or something like that.